### DISCUSSION TITLE
Buidling from source on MacOs Monterey

### DISCUSSION THREAD

**Discussion by jcplajujula** [OpenPrinting/ipp-usb#58] [General]
Hello, 

I'd like to use a 2006 Pixma USB Printer on a 2022 mac.

Do i have any chance of success building ipp-usb on monterey ?

**Comment by alexpevzner** [comment]
AFAIK, IPP over USB is supported on MacOS natively. So what is the purpose to port `ipp-usb` there?

`ipp-usb` depends on Avahi (daemon and client library) and libusb. Avahi is used for MDNS (A.K.A. DNS-SD, Bonjour, Randezvous, Zeroconf...) publishing of device, and libusb is used for the USB access.

If both components are available on MacOS, in theory `ipp-usb` should work there. However, I wonder how Avahi can coexist with MacOS native MDNS infrastructure.
